oauth-keycloak-authorizer/src/main/java/io/strimzi/kafka/oauth/server/authorizer/KeycloakAuthorizer.java

```
@@ -85,13 +89,74 @@ public void configure(Map<String, ?> configs) {
         if (configuration.isDelegateToKafkaACL()) {
             delegate = instantiateStandardAuthorizer();
             delegate.configure(configs);
+            initPluginMetricsIfNeeded();
         }
 
         if (log.isDebugEnabled()) {
             log.debug("Configured " + this + " using " + singleton);
         }
     }
 
+    /**
+     * Initialise and set PluginMetrics on StandardAuthorizer if it implements Monitorable
+     * Use of reflection is needed since the classes are only available starting with Kafka 4.1.0
+     * Once this class implements Monitorable the use of reflection will no longer be necessary, and the passed-in
+     * PluginMetrics object will be passed on to StandardAuthorizer.withPluginMetrics().
+     */
+    private void initPluginMetricsIfNeeded() {
+        Class<?> monitorableClass = null;
+        try {
+            // Check if StandardAuthorizer implements Monitorable, which can only be true if running on Kafka 4.1.0+
+            monitorableClass = Class.forName("org.apache.kafka.common.metrics.Monitorable");
+        } catch (Exception e) {
+            log.debug("Monitorable class not present. PluginMetrics initialisation skipped on StandardAuthorizer");
+        }
+        try {
+            if (monitorableClass != null && monitorableClass.isAssignableFrom(delegate.getClass())) {
+                Class<?> pluginClass = Class.forName("org.apache.kafka.common.internals.Plugin");
+                Metrics metrics = Services.getInstance().getMetrics().getKafkaMetrics();
+                Method method = pluginClass.getMethod("wrapInstance", Object.class, Metrics.class, String.class, String.class, String.class);
+                pluginMetrics = method.invoke(null, delegate, metrics, "authorizer.class.name", "role", determineRole());
+            }
+        } catch (Throwable e) {
+            log.warn("Failed to initialise PluginMetrics on StandardAuthorizer", e);
```

Isn't it fatal if we reach this point?

---

Modified so it fails fast.